### PR TITLE
Use bigger fixture tree for distributed tests

### PR DESCRIPTION
**Pull request by shcheklein**
# Update by @0x2b3bfa0 

## `main` ← #990 ← [#4]

Code rebased and fixed thanks to https://github.com/iterative/studio/pull/11439.

# Original text by @shcheklein 

> A follow up from https://github.com/iterative/studio/pull/10211
>
> Since the default tree fixture is too small `UDFDistributor` was smart enough to run a single worker (despite us passing 2 as UDF parameter in tests).
>
> I don't know if that was always the case or not, but atm it means we are testing some edge case (a single worker running as part of the `LocalWorkerProcess` I believe) and don't really launch other workers (that takes a different path).
> 
> Introducing this bigger fixture reproduced the issue (one of them) in production - shutdown getting stuck. I'm looking into this (unless someone has some immediate ideas) and I'll fix other tests (unless I hit some walls).
> 
> Thanks @dtulga for pointing me to these tests.

### CODE REVIEW

**Comment by mattseddon** [comment]
If you ever get back to this can you please move the tests from DatasetQuery to DataChain

**Comment by skshetry** [comment]
I'll suggest closing this for now, and reopening when you get back to this. It has been opened for quite a while now.

**Comment by skshetry** [comment]
Please reopen this when you get back to it. I am closing for now. :)

**Comment by shcheklein** [comment]
@skshetry that's fine. There are some important insights (it would take some time to find this for someone who is trying to fix it), I hope people will find it.

**Comment by 0x2b3bfa0** [comment]
Sorry for all the force-push noise above: I found a bout in GitHub where it wasn't picking up any of the commits I pushed, and it was closing the pull request with 0 commits.

**Comment by 0x2b3bfa0** [comment]
Extracted the worker fixture to a separate pull request (https://github.com/iterative/datachain/pull/990) because they're completely unrelated and independently useful changes.

File: tests/func/test_datachain.py [hunk 1]
```diff
@@ -780,7 +781,7 @@ def name_len(name):
     for r in dc.collect():
         count += 1
         assert len(r[0]) == r[1]
-    assert count == 7
+    assert count == 7  # FIXME: update after merge
```
Comment (0x2b3bfa0) [inline tests/func/test_datachain.py:784]: Note: need to update this test.

**Review by 0x2b3bfa0 — APPROVED** [review]
Looks very good to me!  

<sub>(Seriously, thanks a lot for the original code and research, @shcheklein)</sub>

File: tests/func/test_datachain.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 from datachain.query.dataset import QueryStep
 from tests.utils import (
     ANY_VALUE,
+    LARGE_TREE,
```
Comment (0x2b3bfa0) [inline tests/func/test_datachain.py:34]: If it wasn't serious code, I would have renamed `LARGE_TREE` to `SEQUOIA`
